mdpow/fep.py

```diff
@@ -252,13 +252,19 @@ def load(cfg, section):
         cfg_get = {float: cfg.getfloat,
                    int: cfg.getint,
                    str: cfg.getstr,    # literal strings, no conversion of None (which we need for the MDP!)
+                                       # CHECK: THIS IS LIKELY NOT GUARANTEED ANYMORE since getstr == get
                    list: cfg.getarray  # numpy float array from list
                    }
         def getter(type, section, key):
+            value = cfg_get[type](section, key)
             try:
-                return cfg_get[type](section, key)
-            except NoOptionError:
-                return None
+                # for empty list [] and array array([])
+                if len(value) == 0:
+                    value = None
+            except TypeError:
+                pass
```

Comment:
How does this exception get hit, and occurs if there is a key error with value?

Reply:
I had to think about it for a moment and convince myself that this actually works as I thought it would. The line
```python
value = cfg_get[type](section, key)
```
can raise `NoSectionError` but I think we don't care about this here (???). But the way that the parser works (again) is that missing options do NOT raise any errors but are substituted with "empty" values, namely None, [], or np.array([]) so that one can easily extend the file format. NoOptionError didn't work anymore since we switched to the yaml file format instead of ini file and the yaml parser does not raise NoOptionError... we just never noticed because we didn't have tests.

So catching the `TypeError` is just saying "if you take to try `len(x)` where `x` is something that is not a sequence then you'll just use `x`... if it's already None, fine, otherwise we use its value.

Nevertheless, there's an untested line in there.... I'll see if I can test that one, too.